### ISSUE TITLE
Unexpected token '=' version 3.4.0

### ISSUE THREAD

**Issue by udhayakumargururaj** [jsdoc3/jsdoc#1246]
Hi,
I am getting this error when running JSDoc.

could you share your suggestions on this.

**Comment by hegemonic** [comment]
Can you provide a source file that causes this error to occur?

**Comment by udhayakumargururaj** [comment]
Thanks for your reply, please find below this line of code cause that issue.
 static propTypes = {
        id: PropTypes.string,
        name: PropTypes.string,
        checked: PropTypes.boolean,
  }

development using ReactJS. when encountering the '=' sign it throws an error.

**Comment by shiftie** [comment]
+1, ES6 class properties doesn't seem supported as well
`
export default class MyClass{
    prop1 = 10000;
    prop2 = 768;

    constructor() {
       ....
`

**Comment by hegemonic** [comment]
I don't think either of those examples are valid ES2015 code. Neither [Espree](https://github.com/eslint/espree/) (the parser JSDoc uses) nor [Esprima](http://esprima.org/) can parse those two examples.

If you're using Babel to process your code, you could try using the [jsdoc-babel plugin](https://www.npmjs.com/package/jsdoc-babel) with JSDoc. Otherwise, I'm not sure what to tell you.

**Comment by udhayakumargururaj** [comment]
Yes, we are using ES6 implementation and using babel for process our code. let me check adding  jsdoc-babel plugin.

**Comment by tathagatbanerjee** [comment]
@hegemonic: tip to use the jsdoc-babel plugin was super useful.

**Comment by NogsMPLS** [comment]
I was able to get JSDoc to handle methods class fields using the arrow function using the `jsdoc-babel` plugin with the `transform-class-properties`. 

*HOWEVER* there is **one weird trick** I needed to do in order for the arrow function definitions to be populated in JSDoc output.

```js
class Foo {
    constructor(){
         // ...code
    }
    /**
     * @memberof Foo
     * @function bar    <--- define function name
     * @description
     *  Does Bar
     */
    bar = () => {
        // ...code
    }

    /**
     * @memberof Foo
     * @description
     *  Does Baz
     */
   baz() { .              <--- for whatever reason, a 'regular' class method must be at the END of the class
        // ...code
   }
}
```

So basically, the arrow function version needed a `@function name` in order to to show it in JSDoc, and the class *had* to have a "regular" method style as the last method. (or more technically all arrow function class properties needed to be above it.)

I'm assuming this is needed because of how babel-jsdoc transforms the code before JSDoc sees it, but I'm not 100% sure. Just posting this in hopes that this maybe helps someone else, at least until the class fields proposal reaches stage 4 and gets into JS proper.

There might be another way to handle this, but this is how I solved it. ;P
